Source/Csla.Blazor.Test/ViewModelSaveAsyncErrorTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿using Csla.Blazor.Test.Fakes;
+using Csla.Test;
+using Csla.TestHelpers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Csla.Blazor.Test
+{
+  [TestClass]
+  public class ViewModelSaveAsyncErrorTests
+  {
+    private static TestDIContext _testDIContext;
+
+    [ClassInitialize]
+    public static void ClassInitialize(TestContext context)
+    {
+      _testDIContext = TestDIContextFactory.CreateDefaultContext();
+    }
+
+    [TestMethod]
+    public async Task SavingFailure_ErrorEventIsInvoked()
+    {
+      // Arrange
+      var person = GetValidFakePerson();
+      var appCntxt = _testDIContext.CreateTestApplicationContext();
+      var vm = new ViewModel<FakePerson>(appCntxt)
+      {
+        ManageObjectLifetime = true,
+        Model = person,
+      };
+      SetupErrorHandler(vm);
+      TestResults.Add("FakePerson", string.Empty);
+      // The second insert with the same key will result in an exception, so the insert will fail
+
+      // Act
+      await vm.SaveAsync();
+
+      // Assert
+      Assert.AreEqual("Invoked", TestResults.GetResult("ErrorHandler"));
+      Assert.IsNotNull(vm.Exception);
+      Assert.IsNotNull(vm.ViewModelErrorText);
+    }
+
+    [TestMethod]
+    public async Task SavingSuccess_ErrorEventIsNotInvoked()
+    {
+      // Arrange
+      var person = GetValidFakePerson();
+      var appCntxt = _testDIContext.CreateTestApplicationContext();
+      var vm = new ViewModel<FakePerson>(appCntxt)
+      {
+        ManageObjectLifetime = true,
+        Model = person,
+      };
+      SetupErrorHandler(vm);
+
+      // Act
+      await vm.SaveAsync();
+
+      // Assert
+      Assert.AreEqual("Inserted", TestResults.GetResult("FakePerson"));
```

Comment:
There is problem using `TestResults` for this test. It's probably due to multiple `ConfigureAwait(false)` that `AsyncLocal` is sometimes addressed from different execution contexts.
I'd wire a dedicated variable in the test and a handler to update variable.
Thoughts?